File: Excel/682 Challenge.py
import pandas as pd

path = "682 Aggregation at Order No Level.xlsx"

input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows=9)
test = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=14).rename(columns=lambda col: col.replace('.1', ''))

input = input.assign(Order_No=input['Order No'].str.split(', ')).explode('Order_No')
result = (input.assign(Amount_pc=input['Amount'] / input.groupby('Name')['Amount'].transform('size'))
         .groupby('Order_No', as_index=False)
         .agg(Names=('Name', lambda x: ', '.join(sorted(set(x)))), Amount=('Amount_pc', 'sum'))
         .sort_values('Order_No')
         .astype({'Order_No': 'int64', 'Amount': 'int64'}))

# Almost equal one field has different sorting of names

File: Excel/600-699/682/682 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/682 Aggregation at Order No Level.xlsx"
input = read_excel(path, range = "A2:C10")
test  = read_excel(path, range = "E2:G15")

result = input %>%
  separate_rows(`Order No`, sep = ", ") %>%
  mutate(`Order No` = as.numeric(`Order No`), 
         Amount_pc = Amount / n(), .by = Name) %>%
  summarise(Names = paste(unique(Name), collapse = ", "), 
            Amount = sum(Amount_pc, na.rm = TRUE), .by = `Order No`) %>%
  arrange(`Order No`)

# all equal except one field has different sorting of names.